.github/workflows/ci.yml

```
@@ -101,18 +127,27 @@ jobs:
+      - uses: qltysh/qlty-action/coverage@v2
+        with:
+          token: ${{ secrets.QLTY_COVERAGE_TOKEN }}
+          files: ./lcov.infA
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v5  # Latest version
         with:
-          args: >-
-            -Dsonar.rust.lcov.reportPaths=lcov.info
+          token: ${{ secrets.CODECOV_TOKEN }}  # Bắt buộc cho private repo
+          file: tarpaulin-report.xml  # File XML từ tarpaulin
```

actionlint (1.7.7)

input "file" is not defined in action "codecov/codecov-action@v5". available inputs are "binary", "codecov_yml_path", "commit_parent", "directory", "disable_file_fixes", "disable_safe_directory", "disable_search", "dry_run", "env_vars", "exclude", "fail_ci_if_error", "files", "flags", "gcov_args", "gcov_executable", "gcov_ignore", "gcov_include", "git_service", "handle_no_reports_found", "job_code", "name", "network_filter", "network_prefix", "os", "override_branch", "override_build", "override_build_url", "override_commit", "override_pr", "plugins", "report_code", "report_type", "root_dir", "skip_validation", "slug", "swift_project", "token", "url", "use_legacy_upload_endpoint", "use_oidc", "verbose", "version", "working-directory"

(action)